contrib/ansible/library/skydive_node_list.py

```diff
@@ -0,0 +1,197 @@
+        default: false
+    insecure:
+        description:
+            - ignore SSL certification verification
+        default: false
+    username:
+        description:
+            - username authentication parameter
+    password:
+        description:
+            - password authentication parameter
+    nodes:
+        description:
+            - list of nodes
+        required: true
+
+notes:
+  - The default value of seed may be unsufficient to disambiguate nodes.
+  - Using list reduces the communication overhead for creating nodes.
+
+author:
+    - Sylvain Afchain (@safchain)
+'''
+
+EXAMPLES = '''
+  - name: create tor
+    skydive_node:
+      nodes:
+        - name: 'TOR1'
+          type: "fabric"
+          seed: "TOR1"
+          metadata:
+            Model: Cisco xxxx
+        - name: 'TOR2'
+          type: "fabric"
+          seed: "TOR2"
+          metadata:
+            Model: Juniper yyyy
+'''
+
+RETURN = '''
+UUID:
+    description: The list of UUID of nodes generated
+    type: list
+'''
+
+import os
+import uuid
+
+from ansible.module_utils.basic import AnsibleModule
+
+from skydive.graph import Node
+
+from skydive.websocket.client import NodeAddedMsgType
+from skydive.websocket.client import WSClient
+from skydive.websocket.client import WSClientDefaultProtocol
+from skydive.websocket.client import WSMessage
+
+
+class NodeInjectProtocol(WSClientDefaultProtocol):
+
+    def onOpen(self):
+        module = self.factory.kwargs["module"]
+        params = self.factory.kwargs["params"]
+        result = self.factory.kwargs["result"]
+
+        if module.check_mode:
+            self.stop()
+            return
+
+        try:
+            host = params["host"]
+            nodes = params["nodes"]
+            uuids = []
+            for node in nodes:
+                metadata = dict()
+                md = node.get("metadata", dict())
+                for (k, v) in md.items():
+                    metadata[k] = v
+                metadata["Name"] = node["name"]
+                metadata["Type"] = node["type"]
+                seed = node.get("seed", "")
+                if len(seed) == 0:
+                    seed = "%s:%s" % (node["name"], node["type"])
+                uid = str(uuid.uuid5(uuid.NAMESPACE_OID, seed))
+
+                node = Node(str(uid), host, metadata=metadata)
+                msg = WSMessage("Graph", NodeAddedMsgType, node)
+                self.sendWSMessage(msg)
+                uuids.append(uid)
+            result["UUID"] = uuids
+        except Exception as e:
+            module.fail_json(
+                msg='Error during topology update %s' % e, **result)
+        finally:
+            self.factory.client.loop.call_soon(self.stop)
+
+
+def run_module():
+    module_args = dict(
+        analyzer=dict(type='str', default="127.0.0.1:8082"),
+        ssl=dict(type='bool', default=False),
+        insecure=dict(type='bool', default=False),
+        username=dict(type='str', default=""),
```

Comment:
shouldn't username & passwords be required?

Reply:
no, if you use the noauth backend for skydive.